packages/sanity/src/core/studio/components/navbar/search/contexts/search/reducer.ts

```diff
@@ -535,3 +558,20 @@ function stripRecent(terms: RecentSearch | SearchTerms) {
   }
   return terms
 }
+
+/**
+ * At page boundaries, the Text Search API may sometimes produce duplicate results. This function
+ * deduplicates an array of results based on their ids.
+ */
+function deduplicate(hits: SearchHit[]): SearchHit[] {
```

Comment:
Nice - maybe worth calling out that it will make sure any preexisting duplicated documents will be moved to the later "page"?